realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -33,6 +33,38 @@
     public static final byte FILE_EXCEPTION_KIND_IMCOMPATIBLE_LOCK_FILE = 4;
     public static final byte FILE_EXCEPTION_KIND_FORMAT_UPGRADE_REQUIRED = 5;
 
+    // Initialized by Realm.init(Context)
+    public static void initialize(File namedPipeDirectory) {
+        if (SharedRealm.namedPipeDirectory != null) {
+            // already initialized
+            return;
+        }
+        if (namedPipeDirectory == null) {
+            throw new IllegalArgumentException("'namedPipeDirectory' must not be null.");
+        }
+
+        String namedPipeDirectoryPath = namedPipeDirectory.getAbsolutePath();
```

Comment:
Since `getCanonicalPath()` is a bit heavy operation and the file object is completely under our control, I'd like to leave as it is.